lib/ts/recipe/oauth2client/recipeImplementation.ts

```diff
@@ -24,8 +24,12 @@ export default function getRecipeImplementation(_querier: Querier, config: TypeN
     let providerConfigWithOIDCInfo: ProviderConfigWithOIDCInfo | null = null;
 
     return {
-        getAuthorisationRedirectURL: async function ({ redirectURIOnProviderDashboard }) {
-            const providerConfig = await this.getProviderConfig();
+        getAuthorisationRedirectURL: async function (
+            this: RecipeInterface,
+            redirectURIOnProviderDashboard,
+            userContext
+        ) {
+            const providerConfig = await this.getProviderConfig(userContext);
```

Comment:
This isn't really what I meant: the `getProviderConfig` should be called in `index.ts` and passed to this function. We just don't want to require users to call this.

Reply:
fixed.